ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -1855,11 +1862,134 @@ void connect_module_instantiation_and_alias(short PASS, ast_node_t* module_insta
+                                input_nets_sc->data[sc_spot_input_old] = (void*)verilog_netlist->one_net;
+                            } else if (strstr(full_name, ZERO_GND_ZERO)) {
+                                join_nets(verilog_netlist->zero_net, (nnet_t*)input_nets_sc->data[sc_spot_input_old]);
+                                free_nnet((nnet_t*)input_nets_sc->data[sc_spot_input_old]);
+                                input_nets_sc->data[sc_spot_input_old] = (void*)verilog_netlist->zero_net;
+                            }
+                            /* check if the instantiation pin exists. */
+                            else if ((sc_spot_output = sc_lookup_string(output_nets_sc, full_name)) == -1) {
+                                /* IF - no driver, then assume that it needs to be aliased to move up as an input */
+                                if ((sc_spot_input_new = sc_lookup_string(input_nets_sc, full_name)) == -1) {
+                                    /* if this input is not yet used in this module then we'll add it */
+                                    sc_spot_input_new = sc_add_string(input_nets_sc, full_name);
+
+                                    /* copy the pin to the old spot */
+                                    input_nets_sc->data[sc_spot_input_new] = input_nets_sc->data[sc_spot_input_old];
+                                } else {
+                                    /* already exists so we'll join the nets */
+                                    combine_nets((nnet_t*)input_nets_sc->data[sc_spot_input_old], (nnet_t*)input_nets_sc->data[sc_spot_input_new], verilog_netlist);
+                                    input_nets_sc->data[sc_spot_input_old] = NULL;
+                                }
+                            } else {
+                                /* ELSE - we've found a matching net, so add this pin to the net */
+                                nnet_t* net = (nnet_t*)output_nets_sc->data[sc_spot_output];
+                                nnet_t* in_net = (nnet_t*)input_nets_sc->data[sc_spot_input_old];
+
+                                if ((net != in_net) && (net->combined == true)) {
+                                    /* if they haven't been combined already, then join the inputs and output */
+                                    join_nets(net, in_net);
+                                    in_net = free_nnet(in_net);
+                                    /* since the driver net is deleted, copy the spot of the in_net over */
+                                    input_nets_sc->data[sc_spot_input_old] = (void*)net;
+                                } else if ((net != in_net) && (net->combined == false)) {
+                                    long idx_array_size = 0;
+                                    long* idx_array = NULL;
+                                    long m;
+                                    for (m = 0; m < input_nets_sc->size; m++) {
```

Comment:
Not sure what m represents; for the future (or after this merge) would be good to choose a more meaningful name.